### ISSUE TITLE
"Auto-lock on minimize" should probably also lock on hide (MacOS)

### ISSUE THREAD

**Issue by DarwinAwardWinner** [keeweb/keeweb#497]
Thanks for taking the time to contribute!

I noticed that when the "Auto-lock on minimize" preference is enabled, hiding the app in Mac OS (i.e. pressing Command+H) doesn't cause the database to be locked. I'm not sure if this is intentional, but it seems to run counter to the spirit of this preference, if not the letter. I would suggest that hiding should count as minimizing for the purposes of auto-locking the database.

**Comment by antelle** [comment]
So, the change will be reverted, because it shouldn't lock on auto-hide, e.g. when the window is hidden by auto-type. For now, there's no simple way to distinguish between them, so we'll leave it for later.

**Comment by DarwinAwardWinner** [comment]
One simple way to distinguish between manual window hiding and window hiding as part of an auto-type action could be timing. If auto-type never hides the window for more than, say, 10 seconds, then just wait a minimum of ten seconds after a hide before locking, and cancel the lock if the window is un-hidden before then.

**Comment by antelle** [comment]
Sounds very hackish, probably there's a better way.

**Comment by EnricoMonese** [comment]
What about listening directly for <kbd>Command</kbd>+<kbd>H</kbd> ?

**Comment by DarwinAwardWinner** [comment]
That wouldn't work, since there are many other ways to hide an application.

**Comment by EnricoMonese** [comment]
Yeah that's true

**Comment by EnricoMonese** [comment]
Maybe something like @DarwinAwardWinner said. The hide event is used but autotype sets a flag to stop it. But if left in the background it gets locked by the _lock after some time_ feature

**Comment by antelle** [comment]
Yes, something like this, but not this way as it was described. We have to remember if we called hide ourselves and lock if it was initiated by the user.
